src/integrationtest/java/com/ericsson/ei/utils/AMQPCommunication.java

```diff
@@ -0,0 +1,313 @@
+ * eiffel.net.ssl.enabled=true javax.net.ssl.trustStore=Path
+ * javax.net.ssl.trustStorePassword=Password
+ */
+public class AMQPCommunication {
+    ConnectionFactory factory;
+    Connection connection;
+    Channel channel;
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AMQPCommunication.class);
+
+    /**
+     * AMQPCommunication constructor.
+     *
+     * @param host
+     *            host name
+     * @param port
+     *            port number
+     */
+    public AMQPCommunication(final String host, final int port) {
+        LOGGER.info("Setting up RabbitMQ connection to '{}:{}'", host, port);
+        this.factory = new ConnectionFactory();
+        this.factory.setHost(host);
+        this.factory.setPort(port);
+        useSSL();
+    }
+
+    /**
+     * AMQPCommunication constructor, using provided connection factory.
+     *
+     * @param factory
+     *            connection factory
+     */
+    public AMQPCommunication(final ConnectionFactory factory) {
+        LOGGER.info("Setting up RabbitMQ connection to '{}:{}'", factory.getHost(), factory.getPort());
+        this.factory = factory;
+        useSSL();
+    }
+
+    /**
+     * Send a message to specified message bus.
+     *
+     * @param message
+     *            content to send
+     * @param exchange
+     *            exchange to receive content
+     * @param key
+     *            routing key
+     * @return true if message was sent, false otherwise
+     */
+    public boolean produceMessage(final String message, final String exchange, final String key) {
+        LOGGER.info("Preparing to produce message -> Host: {}, Exchange: {}, RoutingKey: {}\nMessage: {}",
+                factory.getHost() + ":" + factory.getPort(), exchange, key, message);
+        try {
+            connection = factory.newConnection();
+            channel = connection.createChannel();
+            channel.basicPublish(exchange, key, null, message.getBytes());
+            LOGGER.info("Message being sent.");
+            return true;
+        } catch (IOException | TimeoutException e) {
+            LOGGER.error("An error occured when trying to produce the message.\nError: {}", e.getMessage());
+        } finally {
+            if (connection != null) {
+                try {
+                    connection.close();
+                } catch (IOException e) {
+                    LOGGER.error("Failed to close connection.\nError {}", e.getMessage());
+                }
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Consume a single message synchronously.
+     *
+     * @param queue
+     *            queue to get message from
+     * @return received message
+     */
+    public String consumeMessage(final String queue) {
+        LOGGER.info("Preparing to consume message -> Host: {}, Queue: {}", factory.getHost() + ":" + factory.getPort(),
+                queue);
+        try {
```

Comment:
try to create a method or lambda function for try catch finally. This function will take as argument a function pointer to a method to run inside try and a message to be used in the catch section.